ls1intum/Artemis
# [Exam] Code Editor bugfixes for multiple instances

### Checklist

### Motivation and Context
After today's testing session a few bugs were identified. 

### Description
1. Automatic saving after exercises switching threw a false error on the client
2. The taskbar in the instructions was falsely updated by other exercise's participation results causing the instructions of another exercise which was recently committed to be loaded.
3. Navigation bar did not react to exercise participation changes to update the status when no commit was made by the student. See #1850

### Steps for Testing
1. Log in to Artemis
2. Create an exam with multiple programing exercises
3. Start your exam conduction
4. select a file and edit the content. Switch the exercise, without saving
=> No error should be thrown, it should be saved successfully.
5. Submit your changes and change the exercise. Wait a few seconds until the build is finished and check if the taskbar (in the instructions) is displayed correctly.
6. During all changes, the navigation bar should be updated to show the correct status.
